### PR TITLE
Frr 8.2 upgrade

### PR DESCRIPTION
**Pull request by hasan-brcm**
#### Why I did it
Upgrade FRR to version 8.2.2. Build libyang2 required by FRR.

#### How I did it
Update FRR version and tag.

#### How to verify it
Following tests were performed on sonic-vs:
- BGP docker status check
- BGP configuration and session establishment
- Route redistribution and ping
- Issued show commands to check the bgp neighbor and routes
- Checked app-db to ensure bgp routes are installed with correct interface and nexthop.
- Create VRF and check FRR knows the VRF
- Check VRF routes are installed in app-db with correct Vrf name and next-hop
- Establish BGP Evpn session and check if Evpn routes (multicast, mac, prefix) are exchanged and installed correctly in app-db.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

**Comment by lguohan** [comment]
can you also upgrade frr to bullseye?

**Comment by hasan-brcm** [comment]
Build failing due to  issue #10737

**Comment by hasan-brcm** [comment]
> can you also upgrade frr docker to bullseye?
There are no separate changes for buster/bullseye. Both of the builds should use frr 8.2.2 now. I can see it bullseye got built properly:
[2022-05-05T04:12:31.0610350Z [ finished ] [ target/debs/bullseye/frr_8.2.2-sonic-0_amd64.deb ] ](https://dev.azure.com/mssonic/be1b070f-be15-4154-aade-b1d3bfb17054/_apis/build/builds/96048/logs/39)

**Comment by hasan-brcm** [comment]
kvmtest-t1-lag failed with below error:
_##[error]We stopped hearing from agent sonic-build-workers-000EIM. Verify the agent machine is running and has a healthy network connection. Anything that terminates an agent process, starves it for CPU, or blocks its network access can cause this error. For more information, see: https://go.microsoft.com/fwlink/?linkid=846610_

File: sonic-slave-buster/Dockerfile.j2 [hunk 1]
```diff
@@ -90,7 +90,7 @@ RUN apt-get update && apt-get install -y \
         imagemagick \
         ghostscript \
         groff \
-        libpcre3-dev \
+        libpcre2-dev \
```
Comment (lguohan) [inline sonic-slave-buster/Dockerfile.j2:93]: why this change?

Comment (hasan-brcm) [reply sonic-slave-buster/Dockerfile.j2:93]: libpcre2 is required by libyang2 when building buster. This dependency error is not seen in bullseye build, though. That's why the change only in buster.

dpkg-source: info: extracting libyang2 in libyang2-2.0.112
dpkg-source: info: unpacking libyang2_2.0.112.orig.tar.gz
dpkg-source: info: unpacking libyang2_2.0.112-6.debian.tar.xz
/sonic/src/libyang2/libyang2-2.0.112 /sonic/src/libyang2
dpkg-buildpackage: info: source package libyang2
dpkg-buildpackage: info: source version 2.0.112-6
dpkg-buildpackage: info: source distribution unstable
dpkg-buildpackage: info: source changed by Ondřej Surý <ondrej@debian.org>
 dpkg-source --before-build .
dpkg-buildpackage: info: host architecture amd64
dpkg-checkbuilddeps: error: Unmet build dependencies: libpcre2-dev
dpkg-buildpackage: warning: build dependencies/conflicts unsatisfied; aborting
dpkg-buildpackage: warning: (Use -d flag to override.)
make[1]: *** [Makefile:21: /sonic/target/debs/buster/libyang2_2.0.112-6_amd64.deb] Error 3
make[1]: Leaving directory '/sonic/src/libyang2'
[  FAIL LOG END  ] [ target/debs/buster/libyang2_2.0.112-6_amd64.deb ]
make: *** [slave.mk:493: target/debs/buster/libyang2_2.0.112-6_amd64.deb] Error 1
make: *** Waiting for unfinished jobs....
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this functionality.
Makefile.work:332: recipe for target 'buster' failed
make[1]: *** [buster] Error 2
make[1]: Leaving directory '/projects/csg_sonic2/sr407497/sonic_repos/community_0419/sonic-buildimage'
Makefile:30: recipe for target 'target/debs/buster/frr_8.2.2-sonic-0_amd64.deb' failed
make: *** [target/debs/buster/frr_8.2.2-sonic-0_amd64.deb] Error 2

Comment (hasan-brcm) [reply sonic-slave-buster/Dockerfile.j2:93]: Btw, libpcre3 is already present at L#150 in the same file.

**Comment by adyeung** [comment]
@prsunny @yxieca can someone from MSFT review and signoff pls?

File: src/sonic-frr/patch/0006-frr-remove-frr-log-outchannel-to-var-log-frr.log.patch [hunk 2]
```diff
@@ -1,14 +1,8 @@
-From 31464e3fd0f677bc15286e99d4a30e2e33480c53 Mon Sep 17 00:00:00 2001
-From: Ying Xie <ying.xie@microsoft.com>
-Date: Wed, 25 Nov 2020 22:28:43 +0000
-Subject: [PATCH] [frr] remove frr log outchannel to /var/log/frr.log
+From 0417036a3aece862c95111d4646ca7508a028165 Mon Sep 17 00:00:00 2001
+From: Syed Hasan Raza Naqvi <syed.naqvi@broadcom.com>
+Date: Thu, 21 Apr 2022 17:25:02 -0700
+Subject: [PATCH] [PATCH 6/8] [frr] remove frr log outchannel to /var/log/frr.log
 
-SONiC runs frr inside a docker and the logs are sent to base image
```
Comment (lguohan) [inline src/sonic-frr/patch/0006-frr-remove-frr-log-outchannel-to-var-log-frr.log.patch:6]: can you please not remove the original comments for the patch?

Comment (hasan-brcm) [reply src/sonic-frr/patch/0006-frr-remove-frr-log-outchannel-to-var-log-frr.log.patch:6]: Done.

File: src/sonic-frr/patch/0006-frr-remove-frr-log-outchannel-to-var-log-frr.log.patch [hunk 3]
```diff
@@ -1,14 +1,8 @@
-From 31464e3fd0f677bc15286e99d4a30e2e33480c53 Mon Sep 17 00:00:00 2001
-From: Ying Xie <ying.xie@microsoft.com>
```
[the remark this answers is not in the document]
Comment (hasan-brcm) [reply src/sonic-frr/patch/0006-frr-remove-frr-log-outchannel-to-var-log-frr.log.patch:2]: Done.

**Comment by hasan-brcm** [comment]
> can you please keep original comments for all patches and original author here. updating a patch to a new release should not remove the original author, it would be difficult to track why the patch is made.

Done. Updated all of the patches.. kept original author and commit date intact.

**Comment by zhangyanzhao** [comment]
add @abdosi for review.

**Review by abdosi — COMMENTED** [review]
Can we please update PR description of testing done with this new version.

**Review by abdosi — APPROVED** [review]

**Comment by abdosi** [comment]
LGTM

File: rules/libyang2.mk [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+$(LIBYANG2)_SRC_PATH = $(SRC_PATH)/libyang2
+SONIC_MAKE_DEBS += $(LIBYANG2)
+
+LIBYANG2_DEV = libyang2-dev_$(LIBYANG2_FULLVERSION)_$(CONFIGURED_ARCH).deb
+$(eval $(call add_derived_package,$(LIBYANG2),$(LIBYANG2_DEV)))
+
+LIBYANG2_DBG = libyang2-dbgsym_$(LIBYANG2_FULLVERSION)_$(CONFIGURED_ARCH).deb
+$(eval $(call add_derived_package,$(LIBYANG2),$(LIBYANG2_DBG)))
+
+LIBYANG2_CPP = libyang2-cpp1_$(LIBYANG2_FULLVERSION)_$(CONFIGURED_ARCH).deb
```
Comment (liushilongbuaa) [inline rules/libyang2.mk:23]: It seems this deb package didn't exist

**Comment by DavidZagury** [comment]
The submodule on src/sonic-frr/frr is still pointing to hash c69608a which is part of the FRR branch origin/frr/7.5 and is tagged as version frr-7.5.1-s1.

Shouldn't we also update the submodule?
@lguohan
